mypy/test/teststubgen.py

```diff
@@ -804,6 +809,52 @@ def __init__(self, arg0: str) -> None:
             'def __init__(*args, **kwargs) -> Any: ...'])
         assert_equal(set(imports), {'from typing import overload'})
 
+    def test_cython(self) -> None:
+        pyx_source = """
+#cython: binding=True
+
+import typing
+
+def f(path: str, a: int = 0, b: bool = True) -> typing.List[str]:
+    return []
+
+cdef class MyClass(object):
+    def run(self, action: str) -> None:
+        pass
+        """
+
+        package_name = 'cython_test'
+        tmpdir = tempfile.gettempdir()
+        package_dir = os.path.join(tmpdir, package_name)
+        if not os.path.exists(package_dir):
+            os.mkdir(package_dir)
+        pyx = os.path.join(package_dir, '{}.pyx'.format(package_name))
+        with open(pyx, 'w') as pyx_f:
+            pyx_f.write(pyx_source)
+        subprocess.check_output([
+            'cythonize', '-a', '-i', pyx
+        ])
+
+        os.chdir(tmpdir)
+        outfile = os.path.join(tmpdir, 'out')
+        generate_stub_for_c_module('{}.{}'.format(package_name, package_name), outfile)
+        with open(outfile, 'r') as outfile_f:
+            outfile_txt = outfile_f.read()
+
+        assert """
+class MyClass:
+    @classmethod
+    def __init__(self, *args, **kwargs) -> None: ...
+    def run(self, action: str) -> None: ...
+                """.strip() in outfile_txt
+
+        # The exact format of the resulting .pyi is different depending on Python version
+        assert """
+def f(path: str, a: int = ..., b: bool = ...) -> typing.List[str]: ...
+        """.strip() in outfile_txt or """
+def f(path: str, a: int = ..., b: bool = ...) -> List: ...
+        """.strip() in outfile_txt
```

Comment:
Why are there two signatures that we accept?

This check was a bit confusing. I think that it would be easier to read if it was written along these lines:

```
    sig1 = """..."""
    sig2 = """..."""
    assert sig1.strip() in outfile_txt or sig2.strip() in outfile_txt
```

(Also it would look nicer with `textwrap.dedent()`.)